buildSrc/src/main/kotlin/buildlogic.java-conventions.gradle.kts

```diff
@@ -0,0 +1,69 @@
+plugins {
+    `java-library`
+    id("com.diffplug.spotless")
+    id("de.skuzzle.restrictimports")
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven("https://repo.pgm.fyi/snapshots") // Sportpaper & other pgm-specific stuff
+    maven("https://hub.spigotmc.org/nexus/content/repositories/snapshots/") // Spigot repo
+    maven("https://repo.aikar.co/content/groups/aikar/") // aikar repo
+}
+
+dependencies {
+    implementation("tc.oc.pgm:util:0.16-SNAPSHOT") { isTransitive = false }
+    implementation("com.zaxxer:HikariCP:2.4.1") { isTransitive = false }
+    implementation("fr.minuskube.inv:smart-invs:1.2.7") { isTransitive = false }
+
```

Comment:
In the PR for Events (and Ingame), I needed to add `cloud-annotations` as an implementation to avoid `class file for org.apiguardian.api.API$Status not found` warnings; not sure if that is the correct solution there but this is something I also notice here